compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -928,9 +927,56 @@ trait Implicits:
           if normalImports.isEmpty then suggestedImports else normalImports
         end hiddenImplicitsAddendum
 
-        msg(userDefined.getOrElse(
-          em"no implicit argument of type $pt was found${location("for")}"))() ++
-        hiddenImplicitsAddendum
+        val userDefinedParamMessage = paramSymWithMethodTree.flatMap{ (sym, tree) =>
+          userDefinedMsg(sym, defn.ImplicitNotFoundAnnot).map{ rawMsg =>
+            val paramSyms = typeVariablesInScope(sym)
+            val paramTypeRefs = paramSyms.map(_.typeRef.typeConstructor)
+
+            val argTypes = tree match {
+              case TypeApply(Select(qual, _), targs) =>
+                val methodOwnerType = qual.tpe
+                val methodOwner = tree.symbol.owner
+                val methodTypeParams = tree.symbol.paramSymss.flatten.filter(_.isType)
+                val methodTypeArgs = targs.map(_.tpe)
+                paramTypeRefs.map(_.asSeenFrom(methodOwnerType, methodOwner).subst(methodTypeParams, methodTypeArgs))
+              case _ => paramTypeRefs
+            }
+
+            err.userDefinedErrorString(
+              rawMsg,
+              paramSyms.map(_.name.unexpandedName.toString),
+              argTypes
+            )
+          }
+        }
+
+        val userDefinedTypeMessage = userDefinedMsg(pt.typeSymbol, defn.ImplicitNotFoundAnnot).map{ rawMsg =>
```

Comment:
Same here, I would add a comment with an example of an implicitNotFound on a type definition